frontend/src/components/ProductForm/ProductForm.js

```diff
@@ -28,38 +27,59 @@ const ProductForm = ({ match, history }) => {
 
   const onSubmit = async (e) => {
     e.preventDefault();
-    const response = await API.products.post({
-      data: {
-        ...product,
-        slug: slugify(product.title),
-        structure: 'standalone',
-        stockrecords: [
-          {
-            partner: id,
-            partner_sku: product.sku,
-            price_excl_tax: product.price,
-            num_in_stock: product.stock,
-          }
-        ]
-      },
-      config: { headers: { 'Authorization': `Token ${token}` } },
-    });
-    if (response.status === 201) {
-      // if there are images patch them in later because we need form data here
-      if (product._photos.length) {
-        const formData = new FormData();
-        Array.from(product._photos).forEach(item => {
-          formData.append('original', item);
-        });
-        await API.productImages.post(
-          formData,
-          { headers: { 'Authorization': `Token ${token}` } },
-          response.data.id
-        )
-      }
+    if(editId){
+      await API.products.patch({
+        id: editId,
+        data: {
+          ...product,
+          slug: slugify(product.title),
+          structure: 'standalone',
+          stockrecords: [
+            {
+              partner: id,
+              partner_sku: product.sku,
+              price_excl_tax: product.price,
+              num_in_stock: product.stock,
+            }
+          ]
+        },
+        config: { headers: { 'Authorization': `Token ${token}` }},
+      });
       history.push(`/stores/${id}`);
     } else {
-      console.error(response);
+      const response = await API.products.post({
+        data: {
+          ...product,
+          slug: slugify(product.title),
+          structure: 'standalone',
+          stockrecords: [
+            {
+              partner: id,
+              partner_sku: product.sku,
+              price_excl_tax: product.price,
+              num_in_stock: product.stock,
+            }
+          ]
+        },
+        config: { headers: { 'Authorization': `Token ${token}` } },
+      });
+      if (response.status === 201) {
+        // if there are images patch them in later because we need form data here
+        if (product._photos.length) {
+          const formData = new FormData();
+          Array.from(product._photos).forEach(item => {
+            formData.append('original', item);
+          });
+          await API.productImages.post(
+            formData,
+            { headers: { 'Authorization': `Token ${token}` } },
+            response.data.id
```

Comment:
You are not posting the image for the `patch` yet. Try and use this code here for both cases instead of copying it.

Reply:
{"detail":"Method \"PATCH\" not allowed."}

will be added later